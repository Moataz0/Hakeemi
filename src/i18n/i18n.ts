import i18next from 'i18next';
import en from './en';
import ar from './ar';
import { format } from 'date-fns';
import { enGB as English, ar as Arabic } from 'date-fns/locale';

export const resources = {
  en,
  ar,
} as const;

const getDateLocale = (lng?: string) => {
  switch (lng) {
    case 'ar':
      return Arabic;
    case 'en':
    default:
      return English;
  }
};

const i18n = i18next.createInstance();
i18n.init({
  fallbackLng: 'en',
  lng: 'en',
  supportedLngs: ['en', 'ar'],
  defaultNS: 'common',
  resources,
  react: {
    useSuspense: false,
  },
  interpolation: {
    format: (value, pattern, lng) => {
      if (value instanceof Date && pattern) {
        return format(value, pattern, { locale: getDateLocale(lng) });
      }
      return value;
    },
  },
});

declare module 'react-i18next' {
  type DefaultResources = typeof resources['en'];
  interface Resources extends DefaultResources {}
}

export default i18n;
