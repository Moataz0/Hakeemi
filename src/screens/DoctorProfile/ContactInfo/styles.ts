import { StyleSheet } from 'react-native';
import { palette } from '../../../theme/colors';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderTopWidth: 1,
    borderColor: palette.grey.base,
    marginTop: 16,
  },
  textContainer: { flex: 1, padding: 40 },
  divider: {
    height: '100%',
    borderRightWidth: 1,
    borderColor: palette.grey.base,
  },
});
