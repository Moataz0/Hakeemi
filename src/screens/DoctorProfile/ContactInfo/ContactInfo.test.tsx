import React, { FC } from 'react';
import { render } from '@testing-library/react-native';
import ContactInfo from './ContactInfo';
import { I18nextProvider } from 'react-i18next';
import i18n from '../../../i18n';

const wrapper: FC = ({ children }) => {
  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
};
describe('Button component test', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <ContactInfo
        address="Address"
        email="email@test.com"
        phone="+90 545 693 9297"
      />,
      { wrapper },
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
