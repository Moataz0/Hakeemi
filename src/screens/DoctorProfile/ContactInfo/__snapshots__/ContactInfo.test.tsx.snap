// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "borderColor": "rgba(240, 240, 240, 1)",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "marginTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 40,
        },
        Object {
          "borderColor": "rgba(240, 240, 240, 1)",
          "borderRightWidth": 1,
          "height": "100%",
        },
      ]
    }
  >
    <Text
      selectable={true}
      style={
        Object {
          "color": "rgba(144, 144, 144, 1)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "includeFontPadding": false,
          "lineHeight": 18,
          "textAlign": undefined,
        }
      }
      testID="Text"
    >
      Address
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 40,
      }
    }
  >
    <Text
      selectable={true}
      style={
        Object {
          "color": "rgba(144, 144, 144, 1)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "includeFontPadding": false,
          "lineHeight": 18,
          "textAlign": undefined,
        }
      }
      testID="Text"
    >
      +90 545 693 9297
      <Text
        selectable={true}
        style={
          Object {
            "color": "rgba(144, 144, 144, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 18,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        email@test.com
      </Text>
    </Text>
  </View>
</View>
`;
